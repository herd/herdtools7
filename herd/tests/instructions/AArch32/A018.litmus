ARM A018

{
0:R1=x; 1:R1=x;
0:R3=y; 1:R3=y;
}

 P0               | P1            ;
 MOV R0, #1       | MOV R2, #1    ;
 L0:              | STL  R2, [R3] ;
 LDAEX R6, [R1]   | LDA  R0, [R1] ;
 STLEX R4,R0,[R1] | ;
 CMP R4, #0       | ;
 BNE L0           | ;
 LDREX R2, [R3]   | ;
 DMB ISH          | ;

exists(0:R2=0 /\ 1:R0=0)
