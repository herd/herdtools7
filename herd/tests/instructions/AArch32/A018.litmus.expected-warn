Warning: File "./herd/tests/instructions/AArch32/A018.litmus", unrolling limit exceeded, legal outcomes may be missing.
