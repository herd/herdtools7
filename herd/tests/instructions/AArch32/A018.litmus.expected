Test A018 Allowed
States 4
0:R2=0; 1:R0=0;
0:R2=0; 1:R0=1;
0:R2=1; 1:R0=0;
0:R2=1; 1:R0=1;
Loop Ok
Witnesses
Positive: 3 Negative: 9
Flag Assuming-common-inner-shareable-domain
Condition exists (0:R2=0 /\ 1:R0=0)
Observation A018 Sometimes 3 9
Hash=3d5fb7cad4c9bc287ca8875afd5d42ec

